Sources/OpenTelemetrySdk/Metrics/Stable/Aggregation/AdaptingCircularBufferCounter.swift

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+class AdaptingCircularBufferCounter: NSCopying {
+    func copy(with zone: NSZone? = nil) -> Any {
+        let copy = AdaptingCircularBufferCounter(maxSize: maxSize)
+        copy.startIndex = startIndex
+        copy.endIndex = endIndex
+        copy.baseIndex = baseIndex
+        copy.backing = backing.copy() as! AdaptingIntegerArray
+        return copy
+    }
+
+    private static let NULL_INDEX: Int = Int.min
+    public private(set) var endIndex = NULL_INDEX
+    public private(set) var startIndex = NULL_INDEX
+    private var baseIndex = NULL_INDEX
+    private var backing: AdaptingIntegerArray
+    private let maxSize: Int
+
+    init(maxSize: Int) {
+        self.backing = AdaptingIntegerArray(size: maxSize)
+        self.maxSize = maxSize
+    }
+
+    @discardableResult func increment(index: Int, delta: Int64) -> Bool{
+        if self.baseIndex == Int.min {
```

Comment:
Avoid using self when not needed (e.g. an init when the parameters matches the name )